earthfile2llb/converter.go

```diff
@@ -1339,6 +1404,9 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 		if opts.Transient {
 			return pllb.State{}, errors.New("Transient run not supported with LOCALLY")
 		}
+		if len(c.persistentCacheDirs) > 0 {
+			return pllb.State{}, errors.New("CACHE command not supported with LOCALLY")
+		}
```

Comment:
I would make this check within the `Cache` function. I think there are edge cases that this doesn't catch (e.g. LOCALLY, then SAVE ARTIFACT, without RUN).

Reply:
Understood - and I agree this does make more sense if there are edge cases that would be missed the other way

Reply:
Moved this check so it works in call cases (i.e. not just `RUN`)